--- a/net/minecraft/server/PlayerAdvancements.java
+++ b/net/minecraft/server/PlayerAdvancements.java
@@ -108,65 +_,14 @@
    }
 
    private void m_136006_(ServerAdvancementManager p_136007_) {
-      if (Files.isRegularFile(this.f_263766_)) {
-         try (JsonReader jsonreader = new JsonReader(Files.newBufferedReader(this.f_263766_, StandardCharsets.UTF_8))) {
-            jsonreader.setLenient(false);
-            Dynamic<JsonElement> dynamic = new Dynamic<>(JsonOps.INSTANCE, Streams.parse(jsonreader));
-            int i = dynamic.get("DataVersion").asInt(1343);
-            dynamic = dynamic.remove("DataVersion");
-            dynamic = DataFixTypes.ADVANCEMENTS.m_264140_(this.f_135961_, dynamic, i);
-            Map<ResourceLocation, AdvancementProgress> map = f_135959_.getAdapter(f_135960_).fromJsonTree(dynamic.getValue());
-            if (map == null) {
-               throw new JsonParseException("Found null for advancements");
-            }
-
-            map.entrySet().stream().sorted(Entry.comparingByValue()).forEach((p_265663_) -> {
-               Advancement advancement = p_136007_.m_136041_(p_265663_.getKey());
-               if (advancement == null) {
-                  f_135958_.warn("Ignored advancement '{}' in progress file {} - it doesn't exist anymore?", p_265663_.getKey(), this.f_263766_);
-               } else {
-                  this.m_135985_(advancement, p_265663_.getValue());
-                  this.f_135967_.add(advancement);
-                  this.m_264423_(advancement);
-               }
-            });
-         } catch (JsonParseException jsonparseexception) {
-            f_135958_.error("Couldn't parse player advancements in {}", this.f_263766_, jsonparseexception);
-         } catch (IOException ioexception) {
-            f_135958_.error("Couldn't access player advancements in {}", this.f_263766_, ioexception);
-         }
-      }
-
-      this.m_136002_(p_136007_);
-      this.m_135994_(p_136007_);
    }
 
    public void m_135991_() {
-      Map<ResourceLocation, AdvancementProgress> map = new LinkedHashMap<>();
-
-      for(Map.Entry<Advancement, AdvancementProgress> entry : this.f_263740_.entrySet()) {
-         AdvancementProgress advancementprogress = entry.getValue();
-         if (advancementprogress.m_8206_()) {
-            map.put(entry.getKey().m_138327_(), advancementprogress);
-         }
-      }
-
-      JsonElement jsonelement = f_135959_.toJsonTree(map);
-      jsonelement.getAsJsonObject().addProperty("DataVersion", SharedConstants.m_183709_().m_183476_().m_193006_());
-
-      try {
-         FileUtil.m_257659_(this.f_263766_.getParent());
-
-         try (Writer writer = Files.newBufferedWriter(this.f_263766_, StandardCharsets.UTF_8)) {
-            f_135959_.toJson(jsonelement, writer);
-         }
-      } catch (IOException ioexception) {
-         f_135958_.error("Couldn't save player advancements to {}", this.f_263766_, ioexception);
-      }
-
    }
 
    public boolean m_135988_(Advancement p_135989_, String p_135990_) {
+      // Forge: don't grant advancements for fake players
+      if (this.f_135968_ instanceof net.minecraftforge.common.util.FakePlayer) return false;
       boolean flag = false;
       AdvancementProgress advancementprogress = this.m_135996_(p_135989_);
       boolean flag1 = advancementprogress.m_8193_();
@@ -174,11 +_,14 @@
          this.m_136008_(p_135989_);
          this.f_135967_.add(p_135989_);
          flag = true;
+         net.minecraftforge.event.ForgeEventFactory.onAdvancementProgressedEvent(this.f_135968_, p_135989_, advancementprogress, p_135990_, net.minecraftforge.event.entity.player.AdvancementEvent.AdvancementProgressEvent.ProgressType.GRANT);
          if (!flag1 && advancementprogress.m_8193_()) {
+            this.f_135968_.f_19853_.getCraftServer().getPluginManager().callEvent(new org.bukkit.event.player.PlayerAdvancementDoneEvent(this.f_135968_.getBukkitEntity(), p_135989_.bukkit)); // CraftBukkit
             p_135989_.m_138321_().m_9989_(this.f_135968_);
             if (p_135989_.m_138320_() != null && p_135989_.m_138320_().m_14996_() && this.f_135968_.m_9236_().m_46469_().m_46207_(GameRules.f_46153_)) {
                this.f_135962_.m_240416_(Component.m_237110_("chat.type.advancement." + p_135989_.m_138320_().m_14992_().m_15548_(), this.f_135968_.m_5446_(), p_135989_.m_138330_()), false);
             }
+            net.minecraftforge.event.ForgeEventFactory.onAdvancementEarnedEvent(this.f_135968_, p_135989_);
          }
       }
 
@@ -197,6 +_,7 @@
          this.m_136004_(p_135999_);
          this.f_135967_.add(p_135999_);
          flag = true;
+         net.minecraftforge.event.ForgeEventFactory.onAdvancementProgressedEvent(this.f_135968_, p_135999_, advancementprogress, p_136000_, net.minecraftforge.event.entity.player.AdvancementEvent.AdvancementProgressEvent.ProgressType.REVOKE);
       }
 
       if (flag1 && !advancementprogress.m_8193_()) {
@@ -290,7 +_,7 @@
 
    public AdvancementProgress m_135996_(Advancement p_135997_) {
       AdvancementProgress advancementprogress = this.f_263740_.get(p_135997_);
-      if (advancementprogress == null) {
+      if (advancementprogress == null && p_135997_ != null) {
          advancementprogress = new AdvancementProgress();
          this.m_135985_(p_135997_, advancementprogress);
       }
